src/all/pixiv/src/eu/kanade/tachiyomi/extension/all/pixiv/Pixiv.kt

```diff
@@ -0,0 +1,212 @@
+    private fun illustParse(response: Response): PixivIllust =
+        ajaxParse(response)
+
+    override fun popularMangaRequest(page: Int): Request =
+        searchMangaRequest(page, "漫画", FilterList())
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        val results = ajaxParse<PixivSearchResults>(response)
+
+        val mangas = sequence {
+            results.popular?.run {
+                recent?.let { yieldAll(it) }
+                permanent?.let { yieldAll(it) }
+            }
+        }
+            .map(::searchResultParse)
+            .toList()
+
+        return MangasPage(mangas, hasNextPage = false)
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        var includeNonManga = false
+        var ascendingSort = false
+
+        filters.forEach {
+            when (it) {
+                is NonMangaFilter -> includeNonManga = it.state
+                is SortFilter -> ascendingSort = it.state!!.ascending
+                else -> throw IllegalStateException("Filter unrecognized")
+            }
+        }
+
+        val word = URLEncoder.encode(query, "UTF-8")
+        val type = if (includeNonManga) "artworks" else "manga"
+
+        val parameters = mapOf(
+            "mode" to "all",
+            "order" to if (ascendingSort) "date" else "date_d",
+            "p" to page.toString(),
+            "type" to if (includeNonManga) "all" else "manga",
+            "word" to query,
+        )
+
+        return ajax("GET", "/search/$type/$word", parameters)
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        val results = ajaxParse<PixivSearchResults>(response)
+
+        val mangas = results.run { manga ?: illustManga }!!.data!!
+            .asSequence()
+            .filter { it.isAdContainer != true }
+            .map(::searchResultParse)
+            .toList()
+
+        return MangasPage(mangas, hasNextPage = mangas.isNotEmpty())
+    }
+
+    private fun searchResultParse(result: PixivSearchResult) = SManga.create().apply {
+        url = "/artworks/${result.id!!}"
+        title = result.title ?: ""
+        thumbnail_url = result.url
+    }
+
+    override fun latestUpdatesRequest(page: Int): Request =
+        searchMangaRequest(page, "漫画", FilterList())
+
+    override fun latestUpdatesParse(response: Response): MangasPage =
+        searchMangaParse(response)
+
+    override fun mangaDetailsRequest(manga: SManga): Request =
+        illustRequest(manga.url)
+
+    override fun mangaDetailsParse(response: Response) = SManga.create().apply {
+        val illust = ajaxParse<PixivIllust>(response)
+
+        url = "/artworks/${illust.id!!}"
+        title = illust.title ?: ""
+        artist = illust.userName
+        author = illust.userName
+        description = illust.description?.let { Jsoup.parseBodyFragment(it).wholeText() }
+        genre = illust.tags?.tags?.mapNotNull { it.tag }?.joinToString()
```

Comment:
This one is just a suggestion, following it or not is up to you.

```suggestion
        genre = illust.tags?.tags.orEmpty()
            .filterNotNull { it.tag }
            .joinToString { it.tag }
```